Objects/call.c

```diff
@@ -309,6 +309,30 @@ _PyObject_FastCall(PyObject *func, PyObject *const *args, Py_ssize_t nargs)
 }
 
 
+static PyObject*
+_new_dict_if_nonempty(PyObject *mapping)
```

Comment:
I'd consider extracting the `mapping != NULL` check from this function, and renaming to `shallow_copy_dict()`, `copy_kwds`, or something similar.

Reply:
I had the `mapping` check outside the function initially, but it ended up sort of the worst of both worlds; too much boilerplate in the function itself, but still hiding half of what it was doing in the call. I suspect, given the function is `static` and only used in the one place, that under `--enable-optimizations --with-lto` it gets inlined if it provides any benefit, but I'll admit I haven't checked; performance was good, wasn't going to get too picky.

Reply:
On the name, I like `_copy_kwds`, so I'll go with that; I was probably being too descriptive of the what, less so on the purpose.